pkg/cri/resource-manager/control/rdt/flags.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rdt
+
+import (
+	"flag"
+	"github.com/intel/cri-resource-manager/pkg/config"
+	"github.com/intel/cri-resource-manager/pkg/rdt"
+)
+
+// options captures our configurable parameters.
+type options struct {
+	// ResctrlPath is the mount point of the resctrl pseudo-filesystem.
+	ResctrlPath string `json:",omitempty"`
+	// Class is a assigned to actual RDT class map.
+	Classes map[string]string `json:",omitempty"`
```

Comment:
Do we really want to use `omitempty` in the json tags? (the same question applies to all similar occurrences in other files, too)

Reply:
Well, if I understand it correctly, if we don't tag it with `omitempty` the configuration always must explicitly set a value for it otherwise Unmarshal()ling will fail. IMO the biggest problem with that would be that it leaves no nice way to let something to default to a reasonable built-in value: instead of leaving something unset for the default you'd typically have to set it to the golang zero value of its internal type. So generally speaking, AFAICT all configuration fields with a reasonable built-in default should be tagged with `omitempty`.

Reply:
AFAIU the `omitempty`s are only used in marshalling, not unmarshalling. If a field is not present unmarshalling will just use an "empty" value

Reply:
Actually, if the field is not present unmarshalling just leaves it as it was (does not replace with an "empty" value or anything)

Reply:
Ah, ok. Then we could remove it from a lot of places. But I'd prefer to do that with a single commit addressing all misuses of `omitempty`.

Reply:
Fixed (those introduced in this patch set).